src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -172,6 +174,24 @@ class TaskBuilder(app: AppDefinition,
     Some(builder.build -> ports)
   }
 
+  protected def computeDiscoveryInfo(app: AppDefinition): Option[org.apache.mesos.Protos.DiscoveryInfo] = {
+    app.ipAddress match {
+      case Some(IpAddress(_, _, DiscoveryInfo(ports))) if ports.nonEmpty =>
+        val builder = org.apache.mesos.Protos.DiscoveryInfo.newBuilder
+
+        builder.setName(app.id.toHostname)
+        // TODO(gkleiman@) define the default visibility
+        builder.setVisibility(org.apache.mesos.Protos.DiscoveryInfo.Visibility.FRAMEWORK)
+
+        val portsProto = org.apache.mesos.Protos.Ports.newBuilder
+        portsProto.addAllPorts(ports.map(_.toProto).asJava)
+        builder.setPorts(portsProto)
+
+        Some(builder.build)
+      case _ => None
```

Comment:
If I remember correctly, we decided to not fill DiscoveryInfo in step 1 for non network isolated tasks. Please add a TODO that states that.